### DISCUSSION TITLE
PdfiumError: Failed to load document (PDFium: File access error)

### DISCUSSION THREAD

**Discussion by nithinreddyyyyyy** [pypdfium2-team/pypdfium2#244] [Q&A]
Below is the code i've used 

```py
def convert_pdf_to_images(file_path, scale=300 / 72):
    pdf_file = pdfium.PdfDocument(file_path)

    page_indices = [i for i in range(len(pdf_file))]

    renderer = pdf_file.render(
        pdfium.PdfBitmap.to_pil,
        page_indices=page_indices,
        scale=scale,
    )

    final_images = []

    for i, image in zip(page_indices, renderer):
        image_byte_array = BytesIO()
        image.save(image_byte_array, format='jpeg', optimize=True)
        image_byte_array = image_byte_array.getvalue()
        final_images.append(dict({i: image_byte_array}))

    return final_images
```

I've used that function to load the pdf file using streamlit in chrome, but it is returning error like **"pypdfium2._helpers.misc.PdfiumError: Failed to load document (PDFium: File access error)"**

![image](https://github.com/pypdfium2-team/pypdfium2/assets/129744879/c96ec424-f601-41ad-b1dc-4073c2b78a82)

Can anyone please help me with this?

**Comment by mara004** [comment]
This error means "file not found or could not be opened", so it's an integration issue, not a bug with this project.
(Therefore, I'll transfer your report to Discussions.)

Since pypdfium2 has an `input.is_file()` check raising `FileNotFoundError` if the file does not exist, I guess it might be a permission error in this case?

(Apart from that, your `final_images` data structure with `dict({i: image_byte_array})` items is a bit odd. Why don't you just append/index `image_byte_array` directly? The information is the same.)

**Reply by nithinreddyyyyyy** [reply]
Hey! The code is working on jupyter but unable to open the pdf file in the streamlit app. Might be there's some issue with streamlit or chrome.
